sockeye/data_io.py

```diff
@@ -479,24 +533,63 @@ def _assign_to_buckets(self, source_sentences, target_sentences):
         logger.info("fill up mode: %s", self.fill_up)
         logger.info("")
 
+    def _populate_bucket_batch_sizes(self):
+        """
+        Number of sentences and words per batch for each bucket
+        """
+        # Pre-defined bucket batch sizes
+        if self.bucket_batch_sizes is not None:
+            return
+        # Otherwise compute here
+        self.bucket_batch_sizes = [[] for _ in self.buckets]
+        for buck_idx in range(len(self.buckets)):
```

Comment:
can we use enumerate here to get buck_idx and bucket_shape / padded_seq_len jointly?
Probably also average_seq_len